src/System.Text.Encodings.Web/src/System/Text/Unicode/UnicodeHelpers.cs

```diff
@@ -363,7 +299,7 @@ private static int GetScalarValueFromUtf16Slow(char first, char? second, out boo
                 // unmatched surrogate - substitute
                 Debug.Assert(char.IsLowSurrogate(first));
                 wasSurrogatePair = false;
-                wasUnmatchedSurrogate = true;
+                needMoreData = true; // Last character was high surrgate; we need more data.
```

Comment:
This line is incorrect. `needMoreData` should be _false_ if we see a standalone low surrogate character.

Reply:
Thanks. Please verify with the latest commit.